contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -43,11 +57,133 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
+            _currentTick <= _tickUpper + _logProximity
+            
+            ) {
+            // 4.1. If current fixed rate is within bounds, return false (don't rebalance)
+            return (false);
+        } else {
+            // 4.2. If current fixed rate is outside bounds, return true (do rebalance)
+            return (true);
+        }
+    }
+
+    /// @notice Get the nearest tick multiple given a tick and tick spacing
+    function nearestTickMultiple(int24 newTick, int24 tickSpacing) internal pure returns (int24) {
+     return (newTick / tickSpacing + (newTick % tickSpacing >= tickSpacing/2 ? int24(1) : int24(0)) ) * tickSpacing;
+    }
+
+    /// @notice Set new optimimal tick range based on current tick
+    /// @param currentFixedRateWad currentFixedRate which is passed in from a 7-day rolling avg. historical fixed rate.
+    function rebalance (int256 currentFixedRateWad) public returns (int256 newTickLower, int256 newTickUpper) {
+        _requireAtLeastOperator();
+
+        // console.log('my console logs start here:');
+        // console.logInt(PRBMathSD59x18.log2(currentFixedRateWad));
+        // console.logInt(PRBMathSD59x18.log2(1000100000000000000));
+        // console.logInt(-PRBMathSD59x18.div(PRBMathSD59x18.log2(currentFixedRateWad), PRBMathSD59x18.log2(1000100000000000000)));
+        // console.log(PRBMathUD60x18.mul(_sigmaWad, 2000000000000000000));
+        // console.logInt(nearestTickMultiple(_newTickLower, _tickSpacing));
+
+        if (rebalanceCheck()) {
+            // 0. Get tickspacing from vamm
+            // int24 _tickSpacing = _vamm.tickSpacing(_vamm.address());
+
+            // 1. Get the new tick lower
+            // uint256 _newFixedLowerWad = Math.min(Math.max(0, uint256(currentFixedRateWad) - _sigmaWad), _max_possible_lower_bound);
+            uint256 deltaWad = uint256(currentFixedRateWad) - _sigmaWad;
+            console.log(deltaWad);
+            uint256 _newFixedLowerWad =  0;
+            if (deltaWad > 0) {
+                // delta is greater than 0 => choose delta
+                if (deltaWad < _max_possible_lower_bound) {
+                    _newFixedLowerWad = deltaWad;
+                } else {
+                    _newFixedLowerWad = _max_possible_lower_bound;
+                }
+            } else {
+                // delta is less than 0 => choose 0
+                if (_max_possible_lower_bound > 0) {
+                    _newFixedLowerWad = 0;
+                } else {
+                    _newFixedLowerWad = _max_possible_lower_bound;
+                }
+            }
+            // 2. Get the new tick upper
+            console.log(_newFixedLowerWad);
+            uint256 _newFixedUpperWad = _newFixedLowerWad + 2 * _sigmaWad;
+            console.log(_newFixedUpperWad);
+            // 3. Convert new fixed lower rate back to tick
+            int256 _newTickLowerWad = -PRBMathSD59x18.div(PRBMathSD59x18.log2(int256(_newFixedUpperWad)), 
```

Comment:
need to be careful so that this value is not below MIN TICK, with a smart selection of _max_possible_lower_bound in theory we can avoid this scenario but still feels worth having an extra check @0xZenus @Sol3iman ?

Reply:
Agree. I think we should control this via bounds.